docs/adr/0001_cmsis_component_source_structure_recommendations.md

```diff
@@ -0,0 +1,119 @@
+# CMSIS component source structure recommendations
+Date: 2020-08-19
+
+## Status
+Proposed
+
+## Context
+
+[`ARM-software/CMSIS_5`](https://github.com/ARM-software/CMSIS_5) has been ported to Mbed OS. However, it is split across multiple directories and it is not immediately obvious that it comes from that directory. It also makes it difficult to create an independent CMake target can be built as a library that can be optionally be added by end users.
+
+The current proposal is to create a new directory `CMSIS_5/` at the root of the repository that will mirror closely the structure used by `ARM-software/CMSIS_5/` keeping only the directories and files needed by Mbed OS. It will also include Mbed OS specific files and rename some directories in order to work with Mbed OS build tools (see [Decision](#decision) section below). This will result in the removal of the `rtos/source/TARGET_CORTEX/` and `cmsis/` directories.
+
+Additionally it will provide the following advantages:
+* Better maintenance of component
+* Easy creation of an independent CMake target
+* Easy source navigation that mirrors closely the upstream repository
+
+
+## <a name="decision"></a>Decision
+The following directory restructure is recommended:
```

Comment:
This feel more appropriate to be added to README not to the ADR. This will evolve over time (as we adopt CMake and can just leave things be).